chandra_aca/plot.py

```diff
@@ -0,0 +1,286 @@
+    Plot plot field stars in yang and zang on the supplied
+    axes object in place.
+
+    :param ax: matplotlib axes
+    :param stars: astropy.table compatible set of records of agasc entries of stars
+    :param attitude: Quaternion-compatible attitude
+    :param red_mag_lim: faint limit
+    :param bad_stars: boolean mask of stars to be plotted in red
+    """
+    stars = Table(stars)
+    quat = Quaternion.Quat(attitude)
+
+    if bad_stars is None:
+        bad_stars = np.zeros(len(stars), dtype=bool)
+
+    if 'yang' not in stars.colnames or 'zang' not in stars.colnames:
+        # Add star Y angle and Z angle in arcsec to the stars table.
+        # radec2yagzag returns degrees.
+        yags, zags = radec2yagzag(stars['RA_PMCORR'], stars['DEC_PMCORR'], quat)
+        stars['yang'] = yags * 3600
+        stars['zang'] = zags * 3600
+
+    # Initialize array of colors for the stars, default is black
+    colors = np.zeros(len(stars), dtype='S20')
+    colors[:] = 'black'
+
+    colors[bad_stars] = BAD_STAR_COLOR
+
+    if red_mag_lim:
+        # Mark stars with the FAINT_STAR_COLOR if they have MAG_ACA
+        # that is fainter than red_mag_lim but MAG_ACA - a rough mag error
+        # is brighter than that limit.  The rough mag error calculation is
+        # based on the SAUSAGE acq stage 1 check, which uses nsigma of
+        # 3.0, a mag low limit of 1.5, and a random error of 0.26.
+        nsigma = 3.0
+        mag_error_low_limit = 1.5
+        randerr = 0.26
+        caterr = stars['MAG_ACA_ERR'] / 100.
+        error = nsigma * np.sqrt(randerr**2 + caterr**2)
+        error = np.clip(error, a_min=mag_error_low_limit, a_max=None)
+        faint = ((stars['MAG_ACA'] >= red_mag_lim)
+                 & ((stars['MAG_ACA'] - error) < red_mag_lim))
+        # Faint and bad stars will keep their BAD_STAR_COLOR
+        # Only use the faint mask on stars that are not bad
+        faint = faint & ~bad_stars
+        colors[faint] = FAINT_STAR_COLOR
+        # Don't plot those for which MAG_ACA - error is fainter than red_mag_lim
+        # This overrides any that may be 'bad'
+        too_dim_to_plot = ((stars['MAG_ACA'] >= red_mag_lim)
+                            & ((stars['MAG_ACA'] - error) >= red_mag_lim))
+        colors[too_dim_to_plot] = 'none'
+
+    size = symsize(stars['MAG_ACA'])
+    # scatter() does not take an array of alphas, and rgba is
+    # awkward for color='none', so plot these in a loop.
+    for color, alpha in [(FAINT_STAR_COLOR, FAINT_STAR_ALPHA),
+                         (BAD_STAR_COLOR, BAD_STAR_ALPHA),
+                         ('none', 1.0),
+                         ('black', 1.0)]:
+        colormatch = colors == color
+        ax.scatter(stars[colormatch]['yang'],
+                   stars[colormatch]['zang'],
+                   c=color, s=size[colormatch], edgecolor='none',
+                   alpha=alpha)
+
+
+def star_plot(catalog=None, attitude=None, stars=None, title=None, starcat_time=None,
```

Comment:
Now that I've removed the `plot_catalog` public method, should `star_plot` become `plot_star_catalog` as you originally spec'd?